CarpetX/src/schedule.cxx

```diff
@@ -1515,12 +1517,20 @@ void CycleTimelevels(cGH *restrict const cctkGH) {
         }
         // All time levels (except the current) must be valid everywhere for
         // checkpointed groups
-        if (groupdata.do_checkpoint)
-          for (int tl = (ntls == 1 ? 0 : 1); tl < ntls; ++tl)
-            for (int vi = 0; vi < groupdata.numvars; ++vi)
-              error_if_invalid(groupdata, vi, tl, make_valid_all(), []() {
-                return "CycleTimelevels for the state vector";
-              });
+        if (groupdata.do_checkpoint) {
+          for (int tl = (ntls == 1 ? 0 : 1); tl < ntls; ++tl) {
+            for (int vi = 0; vi < groupdata.numvars; ++vi) {
+              // it is only possible to sync time-level zero
+              if(tl == 0 && presync_only) {
```

Comment:
You can pull this `if` statement out of the `vi` loop since you're not accessing `vi`.